sklearn/decomposition/tests/test_truncated_svd.py

```diff
@@ -220,3 +220,79 @@ def test_singular_values():
     rpca.fit(X_hat_rpca)
     assert_array_almost_equal(apca.singular_values_, [3.142, 2.718, 1.0], 14)
     assert_array_almost_equal(rpca.singular_values_, [3.142, 2.718, 1.0], 14)
+
+
+def test_svd_whiten():
+    tsvd = TruncatedSVD(n_components=X.shape[1] - 1, random_state=42,
+                        whiten=True, algorithm='arpack')
+    Xt = tsvd.fit_transform(X)
+    Xt_2 = tsvd.transform(X)
+    assert_allclose(Xt, Xt_2)
+    Xinv = tsvd.inverse_transform(Xt)
+    # Xdense is a sparse array in dense format (check zeros separately)
+    # The below equality is approximate since n_components < n_features
+    mask = Xdense == 0.0
+    assert_allclose(Xinv[mask], 0, atol=0.08)
+    assert_allclose(Xinv[~mask], Xdense[~mask], rtol=0.04)
```

Comment:
This checks that the reconstuction is also valid with whitening (the case without whitening is checked in [another test](https://github.com/rth/scikit-learn/blob/fd2042dbd641290f24d52b0ead77a7d547a58c6f/sklearn/decomposition/tests/test_truncated_svd.py#L63) . Without this test nothing would really have constrained the result of `inverse_transform` with whitening (apart for the PCA comparison). You are right, I will remove these separate zeros / non zeros checks are that's unnecessary/